library/result.lisp

```diff
@@ -102,6 +102,12 @@
         ((Ok x) (f x))
         ((Err e) (Err e)))))
 
+  (define-instance (Bifunctor Result)
+    (define (bimap f g res)
+      (match res
+        ((Ok x) (Ok (g x)))
+        ((Err e) (Err (f e))))))
```

Comment:
Took me a second to realise that while the `Result` name is the same as Rust, the type param order is the same as `Haskell` and that `:b` is the happy value :laughing:

Reply:
Yeah re: `Functor (Result :a)`